server/app.py

```diff
@@ -84,8 +94,36 @@ def update_cache():
         stop_info = get_stop_predictions(group.get('ids'), group.get('operator'), group.get('group_name'))
         new_stops.append(stop_info)
 
+    # ACE Rail Predictions
+    filtered_times = []
+    TIMESTAMP_FORMAT = "%H:%M:%S"
+    for time in fixed_stops:
+        dt = datetime.datetime.strptime(
+            time,
+            TIMESTAMP_FORMAT,
+        ).replace(tzinfo=datetime.timezone.utc)
+
+        # give the parsed departure time a date of today
+        dt = dt.replace(day=cache.updated_at.day, month=cache.updated_at.month, year=cache.updated_at.year)
+        diff = dt - cache.updated_at
+
+        # if the departure appears to be BEFORE the present moment
+        # move the departure time ahead a day.
+        # for example it was 4PM Pacific (11pm UTC) and we observed the next
+        # train (12am UTC) before. this is wrong. so we should interpret the
+        # next train as coming tomorrow in UTC and recalculate the diff
+        if diff.days < 0:
+            dt = dt.replace(day=dt.day + 1)
+            diff = dt - cache.updated_at
+        
+        # if the difference is under 120 minutes, add it
+        if diff.seconds // 60 < 120:
+            filtered_times.append(dt)
+
+    stop_info = Stop([], "ACE Rail", [Prediction("ACE Train", {'San Jose' : filtered_times})] if filtered_times else [])
```

Comment:
can we put the predictions for ACE under the diridon station id?

Reply:
resolved